### PR TITLE
feat: add license expiration tag

### LINKED ISSUE #24379: Tasklist - Show Non-Commerical License information on the front-end
### Description

Implement functionality to show the status of non-commerical licenses on the front end

The [C3 Header component was recently updated](https://add-license-expiration-tags.camunda-composite-components.pages.dev/?path=/story/uishell-c3navigation--non-production-non-commercial-with-expiry-license-tag) with functionality to show the status of the non-commerical license. Two props were added to the component, `isCommercial` and `expiresAt`. These two props map directly to the new fields from the API

The `/license` endpoint [was extended to add two new fields](https://github.com/camunda/camunda/issues/24092), `isCommercial` and `expiresAt`

The C3 Header component will show the non-commerical license tag only when the `isCommercial` field is `false`
- There are a few different cases for this tag, and each one will display different information to the user
  - When `expiresAt` is more than 30 days in the future. Standard information will show. [See here for an example](https://add-license-expiration-tags.camunda-composite-components.pages.dev/?path=/story/uishell-c3navigation--non-production-non-commercial-with-expiry-license-tag)
  - When `expiresAt` is in less than 30 days, a warning should show. And the user should be shown how many days until expiration. [See here for an example](https://add-license-expiration-tags.camunda-composite-components.pages.dev/?path=/story/uishell-c3navigation--non-production-non-commercial-about-to-expire-license)
  - When `expiresAt` is in the past, an expired messaged should how. [See here for an example](https://add-license-expiration-tags.camunda-composite-components.pages.dev/?path=/story/uishell-c3navigation--non-production-non-commercial-license-is-expired)

When `isCommercial` is `true`, the tag for non-commerical will *not* show in the header

### TODO

- Extend the use of the C3 header in Operate to show the new non-commerical license tag
- Data used in the header will come from the `/license` endpoint

### Additional Information

---------------------------------------------------------------------------------------------

### Acceptance Criteria

- [ ] Checked if the specific data needs to be provided for the QA process

### Definition of Ready - Checklist

- [ ] The issue has a meaningful title, description and testable acceptance criteria

### PR DESCRIPTION
**Pull request by OmranAbazid**
## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #24379

### CODE REVIEW

File: tasklist/client/package.json
Comment (OmranAbazid) [inline tasklist/client/package.json:10]: We probably need to update this to patch/minor version before we merge this PR.
Please review it and i will update this before i merge it.

**Review by OmranAbazid — COMMENTED** [review]
Not sure about the e2e tests failures. They do not seem to be related to my changes.

**Comment by vsgoulart** [comment]
@OmranAbazid is it really supposed to have 2 warnings now?
image

**Comment by vsgoulart** [comment]
Btw, I can take this PR now.

I'm just not sure if we have to show these 2 warnings at the same time or not

**Comment by OmranAbazid** [comment]
> Btw, I can take this PR now.
> 
> I'm just not sure if we have to show these 2 warnings at the same time or not

@vsgoulart According to the C3 Navigation examples, we can have both in this case:
https://add-license-expiration-tags.camunda-composite-components.pages.dev/?path=/story/uishell-c3navigation--non-production-non-commercial-with-expiry-license-tag

I also asked in the Slack channel to be sure:
https://camunda.slack.com/archives/C07QZQ9JKU5/p1731326416084259

**Review by vsgoulart — APPROVED** [review]
The failing E2E will be fixed by a PR from me tomorrow
Let's wait to merge this when the C3 library releases the final version with the changes we need

**Pull request merged**
